src/fromager/http_client.py

```diff
@@ -0,0 +1,36 @@
+"""
+This module acts as a HTTP client with the Retry approach to make code more resilient to network hiccups and connection issues
+It has been adopted from: https://www.raelldottin.com/2023/09/enhancing-resilience-in-your-python.html
+"""
+
+import requests
+from requests.adapters import HTTPAdapter, Retry
+
+
+class HTTPClient:
+    # Set up the retries, backoff_factor and status_forcelist
+    def __init__(self, base_url, retries=5, backoff_factor=0.1, status_forcelist=None):
```

Comment:
If base_url isn't used until the call to get(), does it need to be an argument here?